### PR TITLE
Create one service account per cluster instead of reusing the same SA

### LINKED ISSUE #120: `sc remove-gcp-broker` should delete the service account with Broker Operator role
`sc add-gcp-broker` creates a service account and grants Broker Operator role. But, `sc remove-gcp-broker` does not revert this operation.

Sometimes, `sc add-gcp-broker` leads to a broken state, and then there's no way to fix it. What we can do is to make `sc remove-gcp-broker` clean it up.

### PR DESCRIPTION
**Pull request by maqiuyujoyce**
Fixes #120

### CODE REVIEW

File: installer/pkg/cmd/gcp_broker.go [hunk 1]
```diff
@@ -260,6 +262,17 @@ func removeGCPBroker() error {
 		return fmt.Errorf("error deleting broker resources : %v", err)
 	}
 
+	projectID, err := gcp.GetConfigValue("core", "project")
+	if err != nil {
+		return fmt.Errorf("error getting configured project value : %v", err)
+	}
+
+	brokerSAEmail := fmt.Sprintf("%s@%s.iam.gserviceaccount.com", brokerSAName, projectID)
+	err = gcp.RemoveServiceAccountPerms(projectID, brokerSAEmail, brokerSARole)
```
Comment (martinmaly) [inline installer/pkg/cmd/gcp_broker.go]: Is the role assigned in the context of a project?
If I have 2 clusters in my project, add broker to both, and then remove the broker from one ... what happens?

Comment (maqiuyujoyce) [reply installer/pkg/cmd/gcp_broker.go]: Yeah, good catch. This is just like the broker resource, and is assigned in the context of a project. We should not remove the role unless we are sure all the clusters under the project have removed the gcp broker.

So we can't do it unless we have a counter associated with each project.

@mihnjong Could you provide more context with the original problem so we can workaround it?

**Comment by maqiuyujoyce** [comment]
I refactored the commit based on the new comments in #120. Now sc installer will create one service account for each cluster, and it will be able to create an idempotent experience for the users.

Since the cluster name is actually very long while SA name has a maximum length of 30 characters, I decided to use MD5 hash and base32 encoding to construct valid and unique SA names.

For existent users, a warning is printed when running `sc remove-gcp-broker` so that they will know the deprecation of service account _service-catalog-gcp@[project_id].iam.gserviceaccount.com_.

One question is that: Can different users share the same cluster at the same time? If so, it might cause problems.

This change also fixes #125 and partially addresses #38

**Comment by mihnjong** [comment]
Please test this scenario:

(1) `sc install`
(2) users delete the service acccount
(3) `sc install` again

- In this case, SetIamPolicy would apply to the deleted service account, which is not expected behavior.
- We should remove the IAM role followed by adding the IAM role.

Another scenario:

(1) `sc install`
(2) `sc install` many times

- Is `sc install` idempotent? I guess we keep creating a private key, and it will eventually fail? Failing is okay, but we could output the error message that "Warning: please run `sc uninstall` before running `sc install`".

**Pull request merged**